rt_gene_model_training/pytorch/train_model.py

```diff
@@ -0,0 +1,145 @@
+        _left_patch, _right_patch, _headpose_label, _gaze_labels = batch
+
+        angular_out = self.forward(_left_patch, _right_patch, _headpose_label)
+        loss = self._criterion(angular_out, _gaze_labels)
+        angle_acc = self._angle_acc(angular_out[:, :2], _gaze_labels)
+
+        return {'val_loss': loss, "angle_acc": angle_acc}
+
+    def validation_end(self, outputs):
+        _losses = torch.stack([x['val_loss'] for x in outputs])
+        _angles = np.array([x['angle_acc'] for x in outputs])
+        tensorboard_logs = {'val_loss': _losses.mean(), 'val_angle': np.mean(_angles)}
+        return {'val_loss': _losses.mean(), 'log': tensorboard_logs}
+
+    def configure_optimizers(self):
+        _params_to_update = []
+        for name, param in self._model.named_parameters():
+            if param.requires_grad:
+                _params_to_update.append(param)
+
+        _learning_rate = self.hparams.learning_rate
+        _optimizer = torch.optim.Adam(_params_to_update, lr=_learning_rate, betas=(0.9, 0.95))
+        return _optimizer
+
+    @staticmethod
+    def add_model_specific_args(parent_parser, root_dir):
+        return parent_parser
+
+    @pl.data_loader
+    def train_dataloader(self):
+        _train_transforms = None
+        if self.hparams.augment:
+            _train_transforms = transforms.Compose([transforms.RandomResizedCrop(size=(224, 224), scale=(0.85, 1.0)),
+                                                    transforms.RandomRotation(degrees=5),
+                                                    transforms.Resize((224, 224), Image.BICUBIC),
+                                                    transforms.RandomGrayscale(p=0.08),
+                                                    lambda x: x if np.random.random_sample() > 0.08 else x.filter(ImageFilter.GaussianBlur(radius=5)),
+                                                    lambda x: x if np.random.random_sample() > 0.08 else x.filter(ImageFilter.GaussianBlur(radius=8)),
+                                                    transforms.ToTensor(),
+                                                    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])])
+        _data_train = RTGENEH5Dataset(h5_file=h5py.File(self.hparams.hdf5_file, mode="r"),
+                                      subject_list=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15],
+                                      transform=_train_transforms)
+        return DataLoader(_data_train, batch_size=self.hparams.batch_size, shuffle=True, num_workers=self.hparams.num_io_workers)
+
+    @pl.data_loader
+    def val_dataloader(self):
+        _data_validate = RTGENEH5Dataset(h5_file=h5py.File(self.hparams.hdf5_file, mode="r"), subject_list=[16])
+        return DataLoader(_data_validate, batch_size=self.hparams.batch_size, shuffle=True, num_workers=self.hparams.num_io_workers)
+
+
+if __name__ == "__main__":
+    from pytorch_lightning import Trainer
```

Comment:
Installing this dependency will need to be added to the README.

Reply:
done, also a section on how to train using pytorch pathway.